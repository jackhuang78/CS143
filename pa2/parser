#!/bin/sh
java -classpath /afs/ir/class/cs143/lib/java-cup-11a.jar:.:`dirname $0` Parser $*
