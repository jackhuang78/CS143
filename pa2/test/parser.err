"bad.cl", line 19: syntax error at or near OBJECTID = b
"bad.cl", line 23: syntax error at or near OBJECTID = a
"bad.cl", line 31: syntax error at or near OBJECTID = inherts
"bad.cl", line 41: syntax error at or near OBJECTID = qwe
"bad.cl", line 47: syntax error at or near ERROR = [
"bad.cl", line 49: syntax error at or near OBJECTID = asd
"bad.cl", line 54: syntax error at or near ','
"bad.cl", line 55: syntax error at or near OBJECTID = a
"bad.cl", line 67: syntax error at or near OBJECTID = e
Compilation halted due to lex and parse errors
