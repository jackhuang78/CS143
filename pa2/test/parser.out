#2
_program
  #2
  _class
    A
    Object
    "good.cl"
    (
    )
  #4
  _class
    B
    A
    "good.cl"
    (
    )
  #6
  _class
    C
    Object
    "good.cl"
    (
    #7
    _attr
      a
      Int
      #0
      _no_expr
      : _no_type
    )
  #9
  _class
    D
    C
    "good.cl"
    (
    #10
    _attr
      a
      Int
      #0
      _no_expr
      : _no_type
    )
  #14
  _class
    E
    Object
    "good.cl"
    (
    #17
    _attr
      x
      Int
      #0
      _no_expr
      : _no_type
    #18
    _attr
      y
      Int
      #18
      _int
        10
      : _no_type
    #19
    _attr
      b1
      Bool
      #0
      _no_expr
      : _no_type
    #20
    _attr
      b2
      Bool
      #20
      _bool
        1
      : _no_type
    #22
    _method
      zeroArg
      Int
      #22
      _int
        0
      : _no_type
    #23
    _method
      oneArg
      #23
      _formal
        a
        Int
      Int
      #23
      _int
        1
      : _no_type
    #24
    _method
      twoArg
      #24
      _formal
        a
        Int
      #24
      _formal
        b
        Int
      Int
      #24
      _int
        2
      : _no_type
    #25
    _method
      threeArg
      #25
      _formal
        a
        Int
      #25
      _formal
        b
        Int
      #25
      _formal
        c
        Int
      Int
      #25
      _int
        3
      : _no_type
    #27
    _method
      exprTest
      #27
      _formal
        a
        Int
      #27
      _formal
        b
        Int
      #27
      _formal
        c
        Int
      Int
      #28
      _block
        #30
        _assign
          x
          #30
          _object
            a
          : _no_type
        : _no_type
        #31
        _assign
          x
          #31
          _plus
            #31
            _object
              a
            : _no_type
            #31
            _object
              b
            : _no_type
          : _no_type
        : _no_type
        #32
        _assign
          x
          #32
          _sub
            #32
            _object
              a
            : _no_type
            #32
            _object
              b
            : _no_type
          : _no_type
        : _no_type
        #33
        _assign
          x
          #33
          _mul
            #33
            _object
              a
            : _no_type
            #33
            _object
              b
            : _no_type
          : _no_type
        : _no_type
        #34
        _assign
          x
          #34
          _divide
            #34
            _object
              a
            : _no_type
            #34
            _object
              b
            : _no_type
          : _no_type
        : _no_type
        #35
        _assign
          x
          #35
          _neg
            #35
            _object
              a
            : _no_type
          : _no_type
        : _no_type
        #43
        _assign
          b1
          #43
          _bool
            1
          : _no_type
        : _no_type
        #44
        _assign
          b1
          #44
          _bool
            0
          : _no_type
        : _no_type
        #45
        _assign
          b1
          #45
          _comp
            #45
            _bool
              0
            : _no_type
          : _no_type
        : _no_type
        #46
        _assign
          b1
          #46
          _comp
            #46
            _object
              b1
            : _no_type
          : _no_type
        : _no_type
        #47
        _assign
          b1
          #47
          _lt
            #47
            _object
              x
            : _no_type
            #47
            _object
              y
            : _no_type
          : _no_type
        : _no_type
        #48
        _assign
          b1
          #48
          _leq
            #48
            _object
              x
            : _no_type
            #48
            _object
              y
            : _no_type
          : _no_type
        : _no_type
        #49
        _assign
          b1
          #49
          _eq
            #49
            _object
              x
            : _no_type
            #49
            _object
              y
            : _no_type
          : _no_type
        : _no_type
      : _no_type
    )
  #61
  _class
    F
    Object
    "good.cl"
    (
    #62
    _method
      init0
      Int
      #63
      _block
        #64
        _dispatch
          #64
          _new
            F
          : _no_type
          init0
          (
          )
        : _no_type
        #65
        _dispatch
          #65
          _object
            self
          : _no_type
          init0
          (
          )
        : _no_type
      : _no_type
    #68
    _method
      init
      #68
      _formal
        obj
        F
      Int
      #69
      _block
        #70
        _dispatch
          #70
          _new
            F
          : _no_type
          init
          (
          #70
          _new
            F
          : _no_type
          )
        : _no_type
        #71
        _dispatch
          #71
          _object
            self
          : _no_type
          init
          (
          #71
          _new
            F
          : _no_type
          )
        : _no_type
        #72
        _dispatch
          #72
          _object
            self
          : _no_type
          init
          (
          #72
          _new
            F
          : _no_type
          )
        : _no_type
        #73
        _dispatch
          #73
          _object
            self
          : _no_type
          init
          (
          #73
          _new
            F
          : _no_type
          )
        : _no_type
      : _no_type
    #76
    _method
      init2
      #76
      _formal
        obj
        F
      #76
      _formal
        b
        Int
      Int
      #77
      _dispatch
        #77
        _new
          F
        : _no_type
        init2
        (
        #77
        _new
          F
        : _no_type
        #77
        _int
          2
        : _no_type
        )
      : _no_type
    #79
    _method
      init4
      #79
      _formal
        obj
        F
      #79
      _formal
        b
        Int
      #79
      _formal
        d
        Boolean
      Int
      #80
      _static_dispatch
        #80
        _new
          F
        : _no_type
        E
        init4
        (
        #80
        _new
          F
        : _no_type
        #80
        _int
          2
        : _no_type
        #80
        _bool
          0
        : _no_type
        )
      : _no_type
    )
  #85
  _class
    G
    Object
    "good.cl"
    (
    #86
    _method
      myfunc
      Int
      #87
      _let
        a
        Int
        #0
        _no_expr
        : _no_type
        #87
        _let
          b
          Int
          #87
          _int
            2
          : _no_type
          #87
          _let
            z
            Int
            #87
            _int
              999
            : _no_type
            #87
            _let
              c
              String
              #87
              _string
                "abc"
              : _no_type
              #88
              _block
                #89
                _assign
                  a
                  #89
                  _plus
                    #89
                    _object
                      a
                    : _no_type
                    #89
                    _object
                      b
                    : _no_type
                  : _no_type
                : _no_type
              : _no_type
            : _no_type
          : _no_type
        : _no_type
      : _no_type
    )
